core/src/main/java/com/linecorp/armeria/common/MediaTypeSet.java

```diff
@@ -0,0 +1,345 @@
+     *         any valid ranges.
+     */
+    public Optional<MediaType> matchHeaders(Iterable<? extends CharSequence> acceptHeaders) {
+        requireNonNull(acceptHeaders, "acceptHeaders");
+
+        final List<MediaType> ranges = new ArrayList<>(4);
+        for (CharSequence acceptHeader : acceptHeaders) {
+            addRanges(ranges, acceptHeader);
+        }
+        return match(ranges);
+    }
+
+    /**
+     * Finds the {@link MediaType} in this {@link List} that matches one of the media ranges specified in the
+     * specified string.
+     *
+     * @param acceptHeaders the values of the {@code "accept"} header, as defined in
+     *        <a href="https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html">the section 14.1, RFC2616</a>
+     * @return the most preferred {@link MediaType} that matches one of the specified media ranges.
+     *         {@link Optional#empty()} if there are no matches or {@code acceptHeaders} does not contain
+     *         any valid ranges.
+     */
+    public Optional<MediaType> matchHeaders(CharSequence... acceptHeaders) {
+        requireNonNull(acceptHeaders, "acceptHeaders");
+
+        final List<MediaType> ranges = new ArrayList<>(4);
+        for (CharSequence acceptHeader : acceptHeaders) {
+            addRanges(ranges, acceptHeader);
+        }
+        return match(ranges);
+    }
+
+    private Optional<MediaType> match(List<MediaType> ranges) {
+        return ranges.isEmpty() ? Optional.empty()
+                                : match(ranges.toArray(new MediaType[ranges.size()]));
+    }
+
+    /**
+     * Finds the {@link MediaType} in this {@link List} that matches one of the specified media ranges.
+     *
+     * @return the most preferred {@link MediaType} that matches one of the specified media ranges.
+     *         {@link Optional#empty()} if there are no matches or {@code ranges} does not contain
+     *         any valid ranges.
+     */
+    public Optional<MediaType> match(Iterable<MediaType> ranges) {
+        return match(Iterables.toArray(requireNonNull(ranges, "ranges"), MediaType.class));
+    }
+
+    /**
+     * Finds the {@link MediaType} in this {@link List} that matches one of the specified media ranges.
+     *
+     * @return the most preferred {@link MediaType} that matches one of the specified media ranges.
+     *         {@link Optional#empty()} if there are no matches or {@code ranges} does not contain
+     *         any valid ranges.
+     */
+    public Optional<MediaType> match(MediaType... ranges) {
+        requireNonNull(ranges, "ranges");
+
+        if (ranges.length == 0) {
+            return Optional.empty();
+        }
+
+        MediaType match = null;
+        float matchQ = Float.NEGATIVE_INFINITY;    // higher = better
+        int matchNumWildcards = Integer.MAX_VALUE; // lower = better
+        int matchNumParams = Integer.MIN_VALUE;    // higher = better
+        for (MediaType range : ranges) {
```

Comment:
Ah - well since the utility of this class for normal users doesn't seem so great, I think it'd be reasonable to tighten the types, to say List instead of Iterable. Otherwise, iterating sounds fine to me too.